### PR TITLE
initial attempt at enabling no package import change

### PR DESCRIPTION
**Pull request by eordentlich**
still needs documentation, test, lint type checking etc.   The file `python/test_no_import_change.py` can be tried for some basic testing.

```bash
# to run on cpu
CUDA_VISIBLE_DEVICES=0 python python/test_no_import_change.py 0.2
# run on gpu 
CUDA_VISIBLE_DEVICES=0 python -m spark_rapids_ml python/test_no_import_change.py 0.2
# gpu using spark-submit (for cpu just omit the __main__.py)
CUDA_VISIBLE_DEVICES=0 spark-rapids-submit --master local[2] \
python/test_no_import_change.py 0.2
```

### CODE REVIEW

File: python/src/spark_rapids_ml/install.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def _getattr(attr: str) -> Any:
+        frame = sys._getframe()
+        assert frame.f_back
+        calling_path = frame.f_back.f_code.co_filename
+        if any(
+            [
+                (
+                    f"pyspark/ml/{mod_name}" in calling_path
+                    or f"spark_rapids_ml/{mod_name}" in calling_path
```
Comment (lijinf2) [inline python/src/spark_rapids_ml/install.py:52]: Would this always be true? The calling_path must come from either pyspark or spark_rapids_ml I think.
If so, the if condition will always be true.

File: python/test_no_import_change.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+print(f"cvModel.getNumFolds(): {cvModel.getNumFolds()}")
+print(f"cvModel.avgMetrics[0]: {cvModel.avgMetrics[0]}")
+print(f"evaluator.evaluate(cvModel.transform(dataset)): {evaluator.evaluate(cvModel.transform(dataset))}")
+path = tempfile.mkdtemp()
+model_path = path + "/model"
+cvModel.write().save(model_path)
+cvModelRead = CrossValidatorModel.read().load(model_path)
+print(f"cvModelRead.avgMetrics: {cvModelRead.avgMetrics}")
+print(f"evaluator.evaluate(cvModel.transform(dataset)): {evaluator.evaluate(cvModel.transform(dataset))}")
+print(f"evaluator.evaluate(cvModelRead.transform(dataset)): {evaluator.evaluate(cvModelRead.transform(dataset))}")
```
Comment (lijinf2) [inline python/test_no_import_change.py:91]: Is it possible to check whether both the estimator and the model are from spark_rapids_ml (or both come from pyspark.ml)?

**Review by lijinf2 — COMMENTED** [review]
The idea looks good! Some questions come to my mind.

File: python/src/spark_rapids_ml/install.py (hunk 1 above)
Comment (eordentlich) [reply python/src/spark_rapids_ml/install.py:52]: No it is not always true.   the condition will only be true if the import statement or module attribute access to reach here happened in the specified modules (our modules or select pyspark/ml modules).  In these cases we want to use the pyspark.ml version of the class always.   This allows us to subclass pyspark ml classes like in our CrossValidator and also use pyspark classes to validate parameters.

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

**Comment by wbo4958** [comment]
I just executed the `python -m spark_rapids_ml` but showed nothing. Is this expected? or should print the help usage?

**Comment by eordentlich** [comment]
Yes we could print usage in that case.  Good suggestion.

File: python/src/spark_rapids_ml/install.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                )
+                for mod_name in _accelerated_attributes.keys()
+            ]
+        ) or (attr not in _accelerated_attributes[mod_name]):
+            try:
+                attr_val = getattr(_pyspark_modules[mod_name], attr)
+            except:
+                raise AttributeError("No attribute '{attr}'")
```
Comment (wbo4958) [inline python/src/spark_rapids_ml/install.py:69]: `f"No attribute '{attr}'"`

File: python/tests_no_import_change/test_no_import_change.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+schema = "features array<float>, label float"
+
+spark = SparkSession.builder.getOrCreate()
+df = spark.createDataFrame(data, schema=schema)
+df = df.select(array_to_vector(col("features")).alias("features"), "label")
+df.show()
+
+lr_estimator = LogisticRegression()
+print(lr_estimator.setFeaturesCol("features"))
+
```
Comment (wbo4958) [inline python/tests_no_import_change/test_no_import_change.py:55]: I added `lr_estimator.setStandardization(False)` but it ran with error

`AttributeError: 'LogisticRegression' object has no attribute 'setStandardization'. Did you mean: 'getStandardization'?`

No idea why.

Comment (eordentlich) [reply python/tests_no_import_change/test_no_import_change.py:55]: Turns out we never defined setStandardization for spark_rapids_ml LogisticRegression. adding to this PR.

**Comment by eordentlich** [comment]
build

**Comment by eordentlich** [comment]
build

**Review by lijinf2 — APPROVED** [review]

File: python/tests_no_import_change/test_no_import_change.py (hunk 4 above)
Comment (wbo4958) [reply python/tests_no_import_change/test_no_import_change.py:55]: Yeah, In order to make no code change, seems we need to make sure that spark-rapids-ml contains the APIs of spark.ml

Comment (wbo4958) [reply python/tests_no_import_change/test_no_import_change.py:55]: but it could be in a follow up.

Comment (eordentlich) [reply python/tests_no_import_change/test_no_import_change.py:55]: Yes.  Current goal was to eliminate package import change.   We have quite a few gaps to address for full api compatibility.

**Pull request merged**